### ISSUE TITLE
upgrade-interactive should do engine checks before showing the prompt

### ISSUE THREAD

**Issue by StephanBijzitter** [yarnpkg/yarn#5902]
**Do you want to request a *feature* or report a *bug*?**
I want to request an improvement to an existing feature.

**What is the current behavior?**
Given a local installation of NodeJS 10.0.0 and a package file that requires at least NodeJS 10.1.0, upgrade-interactive only shows an error _after_ the user selects which packages they want to update.

<img width="819" alt="screen shot 2018-05-29 at 10 33 39" src="https://user-images.githubusercontent.com/1649903/40647520-56bc6c3e-632c-11e8-9933-a1a69f3a884e.png">

**If the current behavior is a bug, please provide the steps to reproduce.**
* Edit your `package.json` to include an `engine` requirement for `"node": "x"` where `x` is a version greater than your currently installed NodeJS version.
* Run `yarn upgrade-interactive --latest` and select which dependencies to update.
* See the error that says you need at least version `x` of NodeJS.

**What is the expected behavior?**
* Edit your `package.json` to include an `engine` requirement for `"node": "x"` where `x` is a version greater than your currently installed NodeJS version.
* Run `yarn upgrade-interactive --latest` ~and select which dependencies to update~.
* See the error that says you need at least version `x` of NodeJS.

In short, I would like these errors to show up before I spend time researching and selecting dependencies to update, rather than after. Since these checks are not affected by my upgrade choices, this should be possible.

**Comment by rally25rs** [comment]
Also note that the same logic is used for `yarn outdated` and `yarn upgrade`, so some discussion may be needed to decide if an engine check would be done for `yarn outdated` and/or `yarn upgrade` as well.  I think it makes sense for `upgrade` since the upgrade would fail. For `outdated` I think still showing the ones that would fail an engines check would still be useful.

**Comment by 07Gond** [comment]
Hi @rally25rs ! is anyone working on this?
I want an opportunity to work on this issue if anyone is working on it 😃

**Comment by mikegsrv** [comment]
Hello @rally25rs! 

I was checking first-timers issues and this seems to be featured in #7021 (started from #7013). 
Kindly ask you to check and close this as obsolete if so.

Thank you in advance!

BR,
Mikhail
